transform/g2p/transform.py

```diff
@@ -0,0 +1,112 @@
+
+
+import json
+import collections
+
+from transform.g2p.genes import normalize as genes_normalize
+from transform.g2p.features import normalize as features_normalize
+from transform.g2p.environments import normalize as environments_normalize
+from transform.g2p.phenotypes import normalize as phenotypes_normalize
+from transform.g2p.association import normalize as association_normalize
+from transform.g2p.publications import normalize as publication_normalize
+
+import bmeg.ioutils
+from bmeg.util.logging import default_logging
+from bmeg.util.cli import default_argument_parser
+
+from bmeg.edge import HasSupportingReference, GeneFeatureFor, AlleleFeatureFor, PhenotypeOf, EnvironmentFor
+from bmeg.emitter import *  # noqa dynamic class instantiation
+
+files = {}
+
+
+def normalizeAssociations(path):
+    """ create a record from input """
+    input_stream = bmeg.ioutils.reader(path)
+    # create a tuple where:
+    # vertices - dict of vertices to be created
+    # genes ... publications - arrays of edges from association
+    NormalizedAssociation = collections.namedtuple(
+        'NormalizedAssociation',
+        ['vertices', 'genes', 'features', 'environments', 'phenotypes',
+         'publications', 'association'])
+    for line in input_stream:
+        hit = json.loads(line)
+        (hit, genes) = genes_normalize(hit)
+        (hit, features) = features_normalize(hit)
+        (hit, environments) = environments_normalize(hit)
+        (hit, phenotypes) = phenotypes_normalize(hit)
+        (hit, publications) = publication_normalize(hit)
+        (hit, association) = association_normalize(hit)
+        yield NormalizedAssociation(hit, genes, features, environments,
+                                    phenotypes, publications, association)
+
+
+def toGraph(normalized_association, emitter):
+    """ tuple to graph edges and vertexes """
+    na = normalized_association
+    association = na.association
+    emitter.emit_vertex(association)
+    for publication in na.vertices['publications']:
```

Comment:
I think we should leave it to the pubmed transformer to create the publication vertices.

Reply:
agreed